lib/sle2docker/native_image.rb

```diff
@@ -43,9 +43,12 @@ def compute_metadata_file
       match = regexp.match(@image_name)
       match.nil? &&
         raise(DockerTagError,
-              "Docker image #{@image_name} not found. \
-               Run sle2docker list to check which docker images are available.")
-      @metadata     = parse_metadata_file("#{match['metadata_file']}.metadata")
+              "Docker image #{@image_name} not found. "\
+              'Run sle2docker list to check which docker images are available.')
+      @metadata = parse_metadata_file("#{match['metadata_file']}.metadata")
+
+      validate_metadata(@metadata)
+
       @repository   = @metadata['image']['name']
       @tag          = @metadata['image']['tags'][0]
```

Comment:
In case of --tag-with-build the index number should be 1, correct?

Reply:
@M0ses, no because the tag within the binary will always be loaded and this is the one included with the KIWI config.xml file. `--tag-with-build` will retag the image once it has been already loaded with the configured tag, the result is the same images loaded with multiple tags.

Reply:
But I guess the problem with duplicate tags will persist if you do not change the tag in the kiwi file - like here:

```
source-service:~ # sle2docker activate obs-source-service.x86_64-1.0.0-Build17.2.docker
Image 'obs-source-service:1.0.0' has already been activated.
```

or I have a serious misinterpretation of the whole workflow.

Reply:
Oh yes, this is right. Now I understand the implications of it, I think the best option is to just set the @image_id to the tag+build number in case the `--tag-with-build` is configured. I didn't do that when extending the sle2docker to native images because the image binary includes always the tag from the config.xml, so it looked obvious to me that the image_id should use those values, but I did not realitze that it would break the workflow in which you load multiple builds of the same images without removing the old ones first by using `--tag-with-build`.

I'm working on it, thanks for having a look at it. Will provide a second PR for that, as the current one is consistent as it is and nothing needs to be changed in terms of metadata validation.

Reply:
@Moses I just done a new PR with the change to let you activate the same image with different builds when using the `--tag-with-build` flag